src/Servers/Kestrel/Core/src/Internal/Http3/QPack/IntegerEncoder.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack
+{
+    internal static class IntegerEncoder
+    {
+        public static bool Encode(int i, int n, Span<byte> buffer, out int length)
+        {
+            Debug.Assert(i >= 0);
+            Debug.Assert(n >= 1 && n <= 8);
+
+            var j = 0;
+            length = 0;
+
+            if (buffer.Length == 0)
+            {
+                return false;
+            }
+
+            if (i < (1 << n) - 1)
+            {
+                buffer[j] &= MaskHigh(8 - n);
+                buffer[j++] |= (byte)i;
+            }
+            else
+            {
+                buffer[j] &= MaskHigh(8 - n);
+                buffer[j++] |= (byte)((1 << n) - 1);
+
+                if (j == buffer.Length)
+                {
+                    return false;
+                }
+
+                i = i - ((1 << n) - 1);
+                while (i >= 128)
+                {
+                    buffer[j++] = (byte)(i % 128 + 128);
```

Comment:
`i` is asserted to be positive, so you could use `uint` as type, hence the JIT will produce better code for modulus and division (it uses bit-hacks then).

Reply:
Nah, I always appreciate feedback 😄 . I just don't want a ton of nits about code I'm going to change/delete yet.